WalletWasabi.Daemon/Config/PersistentConfig.cs

```diff
@@ -173,4 +192,51 @@ private PersistentConfig MigrateOldDefaultBackendUris()
 
 		return this;
 	}
+
+	private PersistentConfig MigrateP2pToRpcConnection()
+	{
+		if (ConfigVersion >= 2)
+		{
+			return this;
+		}
+
+		static string? GetRpcCredentialString(BitcoinConfig config, string network, string bitcoindatadir) =>
+			( config.GetSettingOrNull("rpcuser", network)
+			, config.GetSettingOrNull("rpcpassword", network)
+			, config.GetSettingOrNull("rpccookiefile", network)
+			) switch
+			{
+				({ } rpcUser, { } rpcPassword, _) => $"{rpcUser}:{rpcPassword}",
+				( _, null, { } cookieFilePath) => $"cookiefile={cookieFilePath}",
+				( _, null, _) => $"cookiefile={bitcoindatadir}/.cookie",
+				_ => null
+			};
+		static EndPoint GetRpcEndpoint(BitcoinConfig config, string network, int defaultPort) =>
+			(config.GetSettingOrNull("rpcbind", network), config.GetSettingOrNull("rpcport", network)) switch
+			{
+				({} host, {} port) when EndPointParser.TryParse($"{host}:{port}", defaultPort, out var endPoint) => endPoint,
+				({} host, null) when EndPointParser.TryParse(host, defaultPort, out var endPoint) => endPoint,
+				(null, {} port) when int.TryParse(port, out var intPort) => new IPEndPoint(IPAddress.Loopback, intPort),
+				_ => new IPEndPoint(IPAddress.Loopback, defaultPort)
+			};
+
+		var defaultBitcoinDataDir = Network.GetDefaultDataFolder("bitcoin");
+		if (string.IsNullOrWhiteSpace(defaultBitcoinDataDir))
+		{
+			return this;
+		}
+		string configPath = Path.Combine(defaultBitcoinDataDir, "bitcoin.conf");
+		var config = BitcoinConfig.Parse(File.ReadAllText(configPath));
```

Comment:
This is wrong, it assumes that this file exists when we are trying to migrate automatically
It makes my Wasabi to crash with error 

`Message: Could not find a part of the path '/Users/trb/.bitcoin/bitcoin.conf'.`